src/api/cardsApi.js

```diff
@@ -5,7 +5,13 @@ function handleResponse(response) {
   throw new Error(response.text());
 }
 
+function invalid(card) {
+  return (!card.text || !card.text.length || card.columnId === null || card.columnId === undefined);
+}
+
 async function saveCard(card) {
+  if (invalid(card)) throw new Error('Invalid card request');
```

Comment:
Maybe instead of throwing the error you can use the text validation from the material ui and make the form with red border like this: 

![Screenshot from 2020-05-20 20-36-02](https://user-images.githubusercontent.com/16908351/82478475-a1b81d00-9ad9-11ea-9507-75f41e75da3b.png)